structs.go

```diff
@@ -370,7 +370,7 @@ type ChannelEdit struct {
 
 	// NOTE: threads only
 
-	Archived            bool `json:"archived,omitempty"`
+	Archived            bool `json:"archived,bool"`
```

Comment:
Haven't seen this JSON tag before, can you link the source where it is described?

Reply:
But according to your description, I think you meant to do this:
```suggestion
	Archived            *bool `json:"archived,omitempty"`
```

Reply:
Yeah sorry, I just copied what's done two lines below for `Locked` without really thinking much. :flushed: 

Using a pointer is probably the better idea here, but will change the user-facing API for `ChannelEdit``. If that is not an issue, let's do it like this.  The Discord API seems to really allow omitting this field, although the official docs do not really indicate this: Note the `boolean` and not `?boolean` for `archived` and other properties [here](https://discord.com/developers/docs/resources/channel#modify-channel-json-params-thread)

Should I also fix the other `bool` properties (`Locked`,`Invitable`, `NSFW`) here while I'm at it?

Reply:
> Should I also fix the other bool properties (Locked,Invitable, NSFW) here while I'm at it?

Adapted to PR to do this.